external/sqlitejs/sql.js

```diff
@@ -318,11 +318,11 @@ function lc(a,b,c){a="string"!==typeof a?x(a):a;try{var d=c?Ub(a,j):Ub(a);C[b>>2
 function nc(a){a/=1E3;if(da&&window.performance&&window.performance.now)for(var b=window.performance.now();window.performance.now()-b<a;);else for(b=Date.now();Date.now()-b<a;);return 0}r._strlen=oc;r._i64Add=pc;function qc(){return 0}function sc(){f("TODO")}
 var T={ha:function(){return sb(k,"/",16895,0)},md:function(a,b,c){c&&v(1==b==(6==c));a={sd:a,type:b,protocol:c,J:k,Ja:{},Hb:[],ya:[],Aa:T.M};b=T.cb();c=sb(T.root,b,49152,0);c.za=a;b=Lb({path:b,k:c,G:Ib("r+"),seekable:l,o:T.o});a.X=b;return a},qc:function(a){a=P[a];return!a||49152!==(a.k.mode&49152)?k:a.k.za},o:{Cc:function(a){a=a.k.za;return a.Aa.Cc(a)},vc:function(a,b,c){a=a.k.za;return a.Aa.vc(a,b,c)},W:function(a,b,c,d){a=a.k.za;d=a.Aa.Yd(a,d);if(!d)return 0;b.set(d.buffer,c);return d.buffer.length},
 write:function(a,b,c,d){a=a.k.za;return a.Aa.be(a,b,c,d)},close:function(a){a=a.k.za;a.Aa.close(a)}},cb:function(){T.cb.fc||(T.cb.fc=0);return"socket["+T.cb.fc++ +"]"},M:{Xa:function(a,b,c){var d;"object"===typeof b&&(d=b,c=b=k);if(d)d._socket?(b=d._socket.remoteAddress,c=d._socket.remotePort):((c=/ws[s]?:\/\/([^:]+):(\d+)/.exec(d.url))||f(Error("WebSocket URL must be in the format ws(s)://address:port")),b=c[1],c=parseInt(c[2],10));else try{var e=r.websocket&&"object"===typeof r.websocket,g="ws:#".replace("#",
-"//");e&&"string"===typeof r.websocket.url&&(g=r.websocket.url);if("ws://"===g||"wss://"===g)g=g+b+":"+c;var h="binary";e&&"string"===typeof r.websocket.subprotocol&&(h=r.websocket.subprotocol);var h=h.replace(/^ +| +$/g,"").split(/ *, */),m=ca?{protocol:h.toString()}:h;d=new (ca?require("ws"):window.WebSocket)(g,m);d.binaryType="arraybuffer"}catch(t){f(new M(J.Lc))}b={$:b,port:c,q:d,Ya:[]};T.M.$b(a,b);T.M.Fd(a,b);2===a.type&&"undefined"!==typeof a.Ba&&b.Ya.push(new Uint8Array([255,255,255,255,112,
+"//");e&&"string"===typeof r.websocket.url&&(g=r.websocket.url);if("ws://"===g||"wss://"===g)g=g+b+":"+c;var h="binary";e&&"string"===typeof r.websocket.subprotocol&&(h=r.websocket.subprotocol);var h=h.replace(/^ +| +$/g,"").split(/ *, */),m=ca?{protocol:h.toString()}:h;d=new (window.WebSocket)(g,m);d.binaryType="arraybuffer"}catch(t){f(new M(J.Lc))}b={$:b,port:c,q:d,Ya:[]};T.M.$b(a,b);T.M.Fd(a,b);2===a.type&&"undefined"!==typeof a.Ba&&b.Ya.push(new Uint8Array([255,255,255,255,112,
```

Comment:
I removed two uses of the "ws" module (which brings in Web Socket support for Node). We don't need them in the browser so it feels safe to remove them.

Reply:
Hm, in my webpack PR, I did require("sql.js") (straight from node_modules), does this mean we wouldn't want to require sql.js because it brings in too much?

Reply:
I'm not sure! I wonder if the NPM version of this is smarter. We're using a really old version so I think it wasn't built in a particularly effective way.